sktime/forecasting/multiplexer.py

```diff
@@ -0,0 +1,173 @@
+        works just like the selected component.
+        It does not provide any further use in that case.
+
+        When used with ForecastingGridSearchCV, MultiplexerForecaster
+        provides an ability to compare different model class
+        performances with each other, just like a model tournament.
+        When ForecastingGridSearchCV is fitted with a MultiplexerForecaster,
+        returned value for the selected_estimator argument of best_params_
+        attribute of ForecastingGridSearchCV, gives the best
+        performing model class among given models provided in component_estimators.
+
+        Parameters
+        ----------
+        component_estimators : list
+            List of (forecaster names, forecaster objects)
+            MultiplexerForecaster switches between these forecasters
+            objects when used with ForecastingGridSearchCV to
+            find the optimal model
+        selected_estimator: str
+            An argument to make a selection among component_estimators.
+            MultiplexerForecaster uses selected_estimator
+            to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexerForecaster will currently
+            forecast with.
+        _forecaster_fit_params: dict
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.component_estimators = component_estimators
+        self.selected_estimator = selected_estimator
+        self._check_components()
+        self._forecaster_fit_params = None
+
+        super(MultiplexerForecaster, self).__init__()
+
+    def _check_components(self):
+        if not isinstance(self.component_estimators, list):
+            raise Exception(
+                "Please provide a list " "for component_estimators composed of tuples"
+            )
+
+        for component in self.component_estimators:
+            name, estimator = component
+            if not isinstance(estimator, _SktimeForecaster):
+                raise Exception(
+                    "Each component has to be an "
+                    "sktime forecaster object. "
+                    "Please check {} with name:{}".format(estimator, name)
+                )
+
+    def _check_fit_params(self, fit_params):
+        if fit_params is None or fit_params == {}:
+            return
+
+        for component in self.component_estimators:
+            name, _ = component
+
+            if name not in fit_params.keys():
+                raise KeyError(
+                    "If you provide fit_params for models "
+                    "dictionary key of fit params need to "
+                    " match the associated component key."
+                )
+
+    def _check_selected_estimator(self):
```

Comment:
please check out the meta-estimator [here](https://github.com/alan-turing-institute/sktime/blob/d4f3292e310670018658f56b239aac08ce240ab1/sktime/base/_meta.py#L13), we need to make sure that parameters are handled correctly too

Reply:
I believe you meant handling of get_params and set_params. I have extended the class with  `_HeterogenousMetaEstimator  `so that both methods work.